### ISSUE TITLE
Windows 11 22H2 Cursors being applied to the wrong slots

### ISSUE THREAD

**Issue by ShayBox** [ful1e5/Bibata_Cursor#154]
I installed all the cursors/pointers using the install.inf file but everyone is in the wrong slot.

![image](https://github.com/ful1e5/Bibata_Cursor/assets/9505196/fcb189e1-e89b-4265-9aa5-46a07b0e92a5)
![image](https://github.com/ful1e5/Bibata_Cursor/assets/9505196/eb8d34e0-52b6-4102-8a99-c8ca2c55073f)
![image](https://github.com/ful1e5/Bibata_Cursor/assets/9505196/362dfdb1-c29c-417f-ad68-7b3b11a6c7aa)
![image](https://github.com/ful1e5/Bibata_Cursor/assets/9505196/8c140df6-e104-46e7-b0ef-fb0160dc6dfd)

I'm on Windows 11 22H2 (22621.2428)

This is an unrelated request and I know the old Bee/Rainbow/Zebra cursors are unsupported, but could you fix [this issue?](https://github.com/ful1e5/Bibata_Cursor_Rainbow/issues/12), they still work otherwise, and don't need any other changes. I really like the animated cursors and want to use them.

**Comment by stanio** [comment]
The `install.inf` appears to have a cursor list in the wrong order:

```ini
[Scheme.Reg]
HKCU,"Control Panel\Cursors\Schemes","%SCHEME_NAME%",,"%10%\%CUR_DIR%\%alternate%,%10%\%CUR_DIR%\%busy%,%10%\%CUR_DIR%\%cross%,%10%\%CUR_DIR%\%dgn1%,%10%\%CUR_DIR%\%dgn2%,%10%\%CUR_DIR%\%grabbing%,%10%\%CUR_DIR%\%handwriting%,%10%\%CUR_DIR%\%help%,%10%\%CUR_DIR%\%horz%,%10%\%CUR_DIR%\%link%,%10%\%CUR_DIR%\%move%,%10%\%CUR_DIR%\%pan%,%10%\%CUR_DIR%\%person%,%10%\%CUR_DIR%\%pin%,%10%\%CUR_DIR%\%pointer%,%10%\%CUR_DIR%\%text%,%10%\%CUR_DIR%\%unavailable%,%10%\%CUR_DIR%\%vert%,%10%\%CUR_DIR%\%work%,%10%\%CUR_DIR%\%zoom-in%,%10%\%CUR_DIR%\%zoom-out%"
```

It appears sorted alphabetically:

```
%10%\%CUR_DIR%\%alternate%
%10%\%CUR_DIR%\%busy%
%10%\%CUR_DIR%\%cross%
%10%\%CUR_DIR%\%dgn1%
%10%\%CUR_DIR%\%dgn2%
%10%\%CUR_DIR%\%grabbing%
%10%\%CUR_DIR%\%handwriting%
%10%\%CUR_DIR%\%help%
%10%\%CUR_DIR%\%horz%
%10%\%CUR_DIR%\%link%
%10%\%CUR_DIR%\%move%
%10%\%CUR_DIR%\%pan%
%10%\%CUR_DIR%\%person%
%10%\%CUR_DIR%\%pin%
%10%\%CUR_DIR%\%pointer%
%10%\%CUR_DIR%\%text%
%10%\%CUR_DIR%\%unavailable%
%10%\%CUR_DIR%\%vert%
%10%\%CUR_DIR%\%work%
%10%\%CUR_DIR%\%zoom-in%
%10%\%CUR_DIR%\%zoom-out%
```

While it should be:

```ini
HKCU,"Control Panel\Cursors\Schemes","%SCHEME_NAME%",,"%10%\%CUR_DIR%\%pointer%,%10%\%CUR_DIR%\%help%,%10%\%CUR_DIR%\%work%,%10%\%CUR_DIR%\%busy%,%10%\%CUR_DIR%\%cross%,%10%\%CUR_DIR%\%text%,%10%\%CUR_DIR%\%handwriting%,%10%\%CUR_DIR%\%unavailable%,%10%\%CUR_DIR%\%vert%,%10%\%CUR_DIR%\%horz%,%10%\%CUR_DIR%\%dgn1%,%10%\%CUR_DIR%\%dgn2%,%10%\%CUR_DIR%\%move%,%10%\%CUR_DIR%\%alternate%,%10%\%CUR_DIR%\%link%,%10%\%CUR_DIR%\%pin%,%10%\%CUR_DIR%\%person%,%10%\%CUR_DIR%\%pan%,%10%\%CUR_DIR%\%grabbing%,%10%\%CUR_DIR%\%zoom-in%,%10%\%CUR_DIR%\%zoom-out%"
```

Listed vertically:

```
%10%\%CUR_DIR%\%pointer%
%10%\%CUR_DIR%\%help%
%10%\%CUR_DIR%\%work%
%10%\%CUR_DIR%\%busy%
%10%\%CUR_DIR%\%cross%
%10%\%CUR_DIR%\%text%
%10%\%CUR_DIR%\%handwriting%
%10%\%CUR_DIR%\%unavailable%
%10%\%CUR_DIR%\%vert%
%10%\%CUR_DIR%\%horz%
%10%\%CUR_DIR%\%dgn1%
%10%\%CUR_DIR%\%dgn2%
%10%\%CUR_DIR%\%move%
%10%\%CUR_DIR%\%alternate%
%10%\%CUR_DIR%\%link%
%10%\%CUR_DIR%\%pin%
%10%\%CUR_DIR%\%person%
%10%\%CUR_DIR%\%pan%
%10%\%CUR_DIR%\%grabbing%
%10%\%CUR_DIR%\%zoom-in%
%10%\%CUR_DIR%\%zoom-out%
```

A quick online search [reveals](https://thebitguru.com/articles/programmatically-changing-windows-mouse-cursors/3):

>   The cursor schemes contain the path to the cursors for the different cursor types as a comma delimited list. Below are the individual pieces.
>
>   -   Arrow, Help, AppStarting, Wait, Crosshair, IBeam, NWPen, No, SizeNS, SizeWE, SizeNWSE, SizeNESW, SizeAll, UpArrow, Hand _(without the leading spaces)_

The given list doesn't include the newer "Pin" and "Person" entries (#133).  Not sure about the last four:

```
%10%\%CUR_DIR%\%pan%
%10%\%CUR_DIR%\%grabbing%
%10%\%CUR_DIR%\%zoom-in%
%10%\%CUR_DIR%\%zoom-out%
```

They get truncated after being saved to the Registry, at least on Windows 10.

**Comment by stanio** [comment]
@ShayBox, you may give one of the packages I've attached to https://github.com/ful1e5/Bibata_Cursor/issues/149#issuecomment-1783941787 a try.

**Comment by ShayBox** [comment]
Those worked! 👍 Though they were installed as duplicates because they lack the `Cursors` suffix like before, without it they match the `Classic` cursors, but don't match the `Bee/Rainbow/Zebra` cursors. I just had to delete the old ones manually.

**Comment by Ember04** [comment]
Any solution for this? I just installed and saw the problem. The order is in a mess

**Comment by stanio** [comment]
@Ember04, any of:

*   If you're using the _Modern_ variants you could give the packages referenced in https://github.com/ful1e5/Bibata_Cursor/issues/154#issuecomment-1784048253 a try;
*   You could try fixing the `install.inf` as suggested in https://github.com/ful1e5/Bibata_Cursor/issues/154#issuecomment-1783938013, before installing:

    ```diff
    --- a\install.inf
    +++ b\install.inf
    @@ -23,7 +23,7 @@
     Scheme.Cur = 10,"%CUR_DIR%"
    
     [Scheme.Reg]
    -HKCU,"Control Panel\Cursors\Schemes","%SCHEME_NAME%",,"%10%\%CUR_DIR%\%alternate%,%10%\%CUR_DIR%\%busy%,%10%\%CUR_DIR%\%cross%,%10%\%CUR_DIR%\%dgn1%,%10%\%CUR_DIR%\%dgn2%,%10%\%CUR_DIR%\%grabbing%,%10%\%CUR_DIR%\%handwriting%,%10%\%CUR_DIR%\%help%,%10%\%CUR_DIR%\%horz%,%10%\%CUR_DIR%\%link%,%10%\%CUR_DIR%\%move%,%10%\%CUR_DIR%\%pan%,%10%\%CUR_DIR%\%person%,%10%\%CUR_DIR%\%pin%,%10%\%CUR_DIR%\%pointer%,%10%\%CUR_DIR%\%text%,%10%\%CUR_DIR%\%unavailable%,%10%\%CUR_DIR%\%vert%,%10%\%CUR_DIR%\%work%,%10%\%CUR_DIR%\%zoom-in%,%10%\%CUR_DIR%\%zoom-out%"
    +HKCU,"Control Panel\Cursors\Schemes","%SCHEME_NAME%",,"%10%\%CUR_DIR%\%pointer%,%10%\%CUR_DIR%\%help%,%10%\%CUR_DIR%\%work%,%10%\%CUR_DIR%\%busy%,%10%\%CUR_DIR%\%cross%,%10%\%CUR_DIR%\%text%,%10%\%CUR_DIR%\%handwriting%,%10%\%CUR_DIR%\%unavailable%,%10%\%CUR_DIR%\%vert%,%10%\%CUR_DIR%\%horz%,%10%\%CUR_DIR%\%dgn1%,%10%\%CUR_DIR%\%dgn2%,%10%\%CUR_DIR%\%move%,%10%\%CUR_DIR%\%alternate%,%10%\%CUR_DIR%\%link%,%10%\%CUR_DIR%\%pin%,%10%\%CUR_DIR%\%person%,%10%\%CUR_DIR%\%pan%,%10%\%CUR_DIR%\%grabbing%,%10%\%CUR_DIR%\%zoom-in%,%10%\%CUR_DIR%\%zoom-out%"
    
     [Wreg]
     HKCU,"Control Panel\Cursors",,0x00020000,"%SCHEME_NAME%"
    @@ -47,7 +47,7 @@
     HKCU,"Control Panel\Cursors",IBeam,0x00020000,"%10%\%CUR_DIR%\%text%"
     HKCU,"Control Panel\Cursors",No,0x00020000,"%10%\%CUR_DIR%\%unavailable%"
     HKCU,"Control Panel\Cursors",SizeNS,0x00020000,"%10%\%CUR_DIR%\%vert%"
    -HKCU,"Control Panel\Cursors",AppStarteng,0x00020000,"%10%\%CUR_DIR%\%work%"
    +HKCU,"Control Panel\Cursors",AppStarting,0x00020000,"%10%\%CUR_DIR%\%work%"
     HKCU,"Control Panel\Cursors",Zoom-in,0x00020000,"%10%\%CUR_DIR%\%zoom-in%"
     HKCU,"Control Panel\Cursors",Zoom-out,0x00020000,"%10%\%CUR_DIR%\%zoom-out%"
     HKLM,"SOFTWARE\Microsoft\Windows\CurrentVersion\Runonce\Setup\","",,"rundll32.exe shell32.dll,Control_RunDLL main.cpl @0,1"
    ```
*   You could set up a "Bibata" cursor theme/scheme manually, as suggested in https://github.com/ful1e5/Bibata_Cursor/issues/137#issuecomment-1731713946

**Comment by ful1e5** [comment]
@ShayBox I hope this issue is resolved in [v2.0.5](https://github.com/ful1e5/Bibata_Cursor/releases/tag/v2.0.5).

**Comment by ful1e5** [comment]
Closing due to inactivity. Can reopen later.
